File: modules/system-checker/android/src/main/java/expo/modules/systemchecker/SystemCheckerModule.kt
package expo.modules.systemchecker

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import android.os.Build

class SystemCheckerModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("SystemChecker")

    // Defines a JavaScript synchronous function that runs the native code on the JavaScript thread.
    Function("getSystem") {
      Build.DEVICE + ", " + Build.VERSION.SDK_INT
    }
  }
}

File: modules/system-checker/ios/SystemCheckerModule.swift
import ExpoModulesCore

public class SystemCheckerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("SystemChecker")

    // Defines a JavaScript synchronous function that runs the native code on the JavaScript thread.
    Function("getSystem") { () -> String in
    "\(UIDevice.current.name), \(UIDevice.current.systemVersion)"
    }
  }
}
